schedule.py

```diff
@@ -136,15 +168,33 @@ def read_input(input_file):
                 section = Section(course_name, section_num, units, semester)
                 sections[section.name] = section
 
+    # create Time objects
+    # Time have start time, end time, list of 1/0 for weekdays, and list of conflicts with all time slots
+    times = []
+    for _, rows in time_tab.iterrows():
+        info = rows.tolist()
+        times.append(Time(start=info[5], end=info[6], weekdays=info[0:5]))
+    for t in times:
+        conflicts = {}
+        for c in times:
+            if t.conflict(c):
+                conflicts[c] = 1
+            else:
+                conflicts[c] = 0
```

Comment:
I don't think you need a dictionary here, just to store a boolean. A `set()` is a better way to go, to just store the conflicting times, and you can then whether two times conflict with `if t1 in t2.conflicts` or similar.